internal/receiver/smartagentreceiver/config.go

```diff
@@ -77,6 +80,23 @@ func mergeConfigs(componentViperSection *viper.Viper, intoCfg interface{}) error
 		delete(allSettings, "endpoint")
 	}
 
+	if metadataClients, containsClients := allSettings["metadataclients"]; containsClients {
+		if clients, isSlice := metadataClients.([]interface{}); isSlice {
+			var mClients []string
+			for _, c := range clients {
+				if client, isString := c.(string); isString {
+					mClients = append(mClients, client)
+				} else {
+					return errMetadataClientValue
+				}
+			}
+			receiverCfg.MetadataClients = &mClients
+		} else {
+			return errMetadataClientValue
+		}
+		delete(allSettings, "metadataclients")
+	}
```

Comment:
What is this code for?

Reply:
Custom unmarshalling of `metadataClients` to a `*[]string`.  In viper unmarshallers `allSettings` is a `map[string]interface{}`.

Reply:
Will add comment